api/institutions/authentication.py

```diff
@@ -86,6 +123,46 @@ def authenticate(self, request):
         suffix = provider['user'].get('suffix')
         department = provider['user'].get('department')
 
+        # Check secondary institutions which uses the SSO of primary ones
+        secondary_institution = None
+        if provider['id'] in INSTITUTION_SHARED_SSO_MAP:
+            switch_map = INSTITUTION_SHARED_SSO_MAP[provider['id']]
+            criteria_type = switch_map.get('criteria')
+            if criteria_type == 'attribute':
+                attribute_name = switch_map.get('attribute')
+                attribute_value = provider['user'].get(attribute_name)
+                if attribute_value:
+                    secondary_institution_id = switch_map.get('institutions').get(attribute_value)
```

Comment:
Minor: less important because it's a defined above rather than a variable param, but `{}.get('a').get('b')` results in an `AttributeError`, whereas `{}.get('a', {}).get('b')` does not.

Reply:
Yep, I didn't do a check since it is defined, but `{}.get('a', {}).get('b')` is a very good suggestion. Let me update this.